### PR TITLE
COCOS-37 - Add computation key to auth headers for events

### PR DESCRIPTION
**Pull request by WashingtonKK**
Adds the computation key as part of the authorization headers.

### CODE REVIEW

File: internal/events/events.go [hunk 1]
```diff
@@ -29,21 +29,23 @@ func New(svc, serverUrl string) Service {
...
-		EventType     string          `json:"event_type"`
-		Timestamp     time.Time       `json:"timestamp"`
-		ComputationID string          `json:"computation_id,omitempty"`
-		Details       json.RawMessage `json:"details,omitempty"`
-		Originator    string          `json:"originator"`
-		Status        string          `json:"status,omitempty"`
+		EventType      string          `json:"event_type"`
+		Timestamp      time.Time       `json:"timestamp"`
+		ComputationID  string          `json:"computation_id,omitempty"`
+		ComputationKey string          `json:"computation_key,omitempty"`
```
Comment (SammyOina) [inline internal/events/events.go:37]: why is computation key part f the json body and also part of the request header

File: manager/service.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ func New(qemuCfg qemu.Config, logger mglog.Logger, eventSvc events.Service, host
 }
 
 func (ms *managerService) Run(ctx context.Context, c *Computation) (string, error) {
-	ms.publishEvent("vm-provision", c.Id, "starting", json.RawMessage{})
+	ms.publishEvent("vm-provision", c.Id, c.Key, "starting", json.RawMessage{})
 	ac := agent.Computation{
```
Comment (SammyOina) [inline manager/service.go:62]: is key copied to ac before being added to agent, this is missing, also do manual tests

**Review by SammyOina — CHANGES_REQUESTED** [review]

File: manager/service.go (hunk 2 above)
Comment (WashingtonKK) [reply manager/service.go:62]: The PR is still draft because it is pending manual tests

File: agent/computations.go [hunk 3]
```diff
@@ -24,6 +24,7 @@ type AgentConfig struct {
 
 type Computation struct {
 	ID              string      `json:"id,omitempty"`
+	Key             string      `json:"-"`
```
Comment (SammyOina) [inline agent/computations.go:27]: agent does not require key as its events are proxied on manager

File: agent/events/events.go [hunk 4]
```diff
@@ -16,8 +16,13 @@ type service struct {
...
 }
 
+type Header struct {
+	Key   string
+	Value string
+}
+
 type Service interface {
-	SendEvent(event, status string, details json.RawMessage) error
+	SendEvent(event, status string, details json.RawMessage, headers []Header) error
```
Comment (SammyOina) [inline agent/events/events.go:25]: create the header on manager side not on agent

File: agent/service.go [hunk 5]
```diff
@@ -78,14 +78,25 @@ func New(ctx context.Context, logger *slog.Logger, eventSvc events.Service) Serv
...
+	header := []events.Header{
+		{
+			Key:   "Authorization",
+			Value: "Bearer " + svc.computation.Key,
+		},
+		{
+			Key:   "Content-Type",
+			Value: "application/json",
+		},
+	}
```
Comment (SammyOina) [inline agent/service.go:91]: agent does not do http comms so this sould be on manager

**Review by SammyOina — CHANGES_REQUESTED** [review]
extend agentevents to map computation keys to computation ids and set headers beore forwarding agent events from manager

File: manager/service.go [hunk 6]
```diff
@@ -44,21 +45,31 @@ type managerService struct {
 	logger   *slog.Logger
 	eventSvc events.Service
 	hostIP   string
-	agents   map[int]string // agent map of vsock cid to computationID.
+	listener *vsock.Listener
```
Comment (SammyOina) [inline manager/service.go:48]: name appropriately

File: manager/service.go [hunk 7]
```diff
@@ -44,21 +45,31 @@ type managerService struct {
...
+	agents   map[int]string    // agent map of vsock cid to computationID.
+	keys     map[string]string // key map of computationID to key.
```
Comment (SammyOina) [inline manager/service.go:50]: combine these structures as they store simillar things

Comment (SammyOina) [inline manager/service.go:50]: combine these structures as they store simillar things

File: manager/service.go [hunk 8]
```diff
@@ -44,21 +45,31 @@ type managerService struct {
...
 }
 
 var _ Service = (*managerService)(nil)
 
 // New instantiates the manager service implementation.
 func New(qemuCfg qemu.Config, logger *slog.Logger, eventSvc events.Service, hostIP string) Service {
+	l, err := vsock.Listen(VsockEventsPort, nil)
```
Comment (SammyOina) [inline manager/service.go:57]: move to agent events

File: manager/service.go [hunk 9]
```diff
@@ -44,21 +45,31 @@ type managerService struct {
...
 		qemuCfg:  qemuCfg,
 		eventSvc: eventSvc,
 		hostIP:   hostIP,
 		logger:   logger,
+		listener: l,
 		agents:   make(map[int]string),
+		keys:     make(map[string]string),
 	}
 	go ms.retrieveAgentLogs()
+	go ms.forward()
```
Comment (SammyOina) [inline manager/service.go:72]: name appropriately

Comment (SammyOina) [inline manager/service.go:72]: name appropriately

File: manager/service.go [hunk 10]
```diff
@@ -114,6 +125,12 @@ func (ms *managerService) Run(ctx context.Context, c *Computation) (string, erro
 	return fmt.Sprintf("%s:%d", ms.hostIP, ms.qemuCfg.HostFwdAgent), nil
 }
 
+func (ms *managerService) GetComputationKey(ctx context.Context, cid int) string {
```
Comment (SammyOina) [inline manager/service.go:128]: unnecessary function

**Review by SammyOina — CHANGES_REQUESTED** [review]

**Pull request closed without merging**